scen_01/scenario_01.py

```diff
@@ -0,0 +1,181 @@
+
+
+def list_cm():
+    """ Return a list of the cm"""
+    response = requests.get(settings.CM_ENDPOINT)
+    try:
+        cms_list = response.json()["cms"]
+    except:
+        raise KeyError('"cms" key does not exist.')
+    cms = dict()
+    for i, cm in enumerate(cms_list):
+        try:
+            pretty_name = cm["pretty_name"]
+            name = cm["name"]
+        except:
+            raise KeyError('"pretty_name" key does not exist.')
+        cms.update({i: [{"name": name}, {"pretty_name": pretty_name}]})
+    return cms
+
+
+def create_task(cm_name: str, is_fake_cm: bool = False):
+    if is_fake_cm:
+        cm_name += "_fake"
+    headers = {
+        "Connection": "keep-alive",
+        "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.114 Safari/537.36 Edg/89.0.774.68",
+        "Content-Type": "application/json",
+        "Accept": "*/*",
+        "Origin": "http://127.0.0.1:7000",
+        "Sec-Fetch-Site": "same-origin",
+        "Sec-Fetch-Mode": "cors",
+        "Sec-Fetch-Dest": "empty",
+        "Referer": "http://127.0.0.1:7000/",
+        "Accept-Language": "fr,fr-FR;q=0.9,en;q=0.8,en-GB;q=0.7,en-US;q=0.6",
+    }
+    data = {
+        "selection": {
+            "type": "FeatureCollection",
+            "features": [
+                {
+                    "type": "Feature",
+                    "properties": {},
+                    "geometry": {
+                        "type": "Polygon",
+                        "coordinates": [
+                            [
+                                [-0.993347, 51.407202],
+                                [-0.411072, 51.41177],
+                                [-0.383606, 51.012603],
+                                [-1.018982, 51.036788],
+                                [-0.993347, 51.407202],
+                            ]
+                        ],
+                    },
+                }
+            ],
+        },
+        "layers": ["gfa_tot_curr_density.tiff"],
+        "parameters": {"factor": 3},
+    }
+    url = urljoin(settings.CM_ENDPOINT, cm_name + "/task")
+    response = requests.post(url=url, headers=headers, json=data)
+    try:
+        dict_task = response.json()
+        try:
+            dict_task["task_id"]
+        except:
+            if "message" in dict_task.keys():
+                raise KeyError(dict_task["message"])
+            else:
+                raise KeyError("Key does not exist : task_id")
+    except json.JSONDecodeError as err:
+        err.msg += ", content received was " + response.text
+        raise err
+    return response.url
+
+
+def get_task(task_url: str):
+    status = "PENDING"
+    while status == "PENDING":
+        try:
+            response = requests.get(url=task_url)
+        except:
+            raise requests.ConnectionError("URL is not valid : {}".format(task_url))
+        dict_task = response.json()
+        status = dict_task["status"]
```

Comment:
I am not sure how we need to do this...
@neodark  can you help us, please?

Reply:
What is this exactly ?
What is this `dict_task` ?
What is returned by response ?
Why do we need an unused `status` variable